src/cluster_writer.rs

```diff
@@ -93,7 +94,10 @@ impl ClusterWriter {
     self.previous_position
   }
 
-  pub fn write_item(&mut self, mut item: Item) -> Result<()> {
+  pub fn write_item<MDT>(&mut self, mut item: Item<MDT>) -> Result<()>
+  where
+    for<'de2> MDT: Deserialize<'de2> + Serialize + PartialEq + Clone + Mergeable + Sized,
```

Comment:
Can we create a single trait to encapsulate all of these?

Reply:
Addressed